tests/robustness/multiclient_test/framework/harness.go

```diff
@@ -207,31 +208,48 @@ func (th *TestHarness) Engine() *engine.Engine {
 
 // Run runs the provided function asynchronously for each of the given client
 // contexts, waits for all of them to finish, and optionally cleans up clients.
-func (th *TestHarness) Run(ctxs []context.Context, cleanup bool, f func(context.Context)) {
-	var wg sync.WaitGroup
-
-	for _, ctx := range ctxs {
-		wg.Add(1)
-
-		go func(ctx context.Context) {
-			f(ctx)
-
-			if cleanup {
-				th.snapshotter.CleanupClient(ctx)
-			}
+func (th *TestHarness) Run( // nolint:thelper
+	ctxs []context.Context,
+	t *testing.T, cleanup bool,
+	f func(context.Context, *testing.T),
+) {
+	t.Helper()
```

Comment:
t.Helper() seems counter-productive here as it will hide the details about which line failed, I would ignore the linter error instead

Reply:
(same thing applies in other places)

Reply:
Good point, thanks for reviewing! I just updated with the suggested changes.